### DISCUSSION TITLE
How to download & run a GitHub version of yfinance

### DISCUSSION THREAD

**Discussion by ValueRaider** [ranaroussi/yfinance#1080] [Development]
There are two different methods to download & run a GitHub version of `yfinance`. Pick one:
1. With PIP
2. Without PIP

**Comment by ValueRaider** [comment]
# With PIP

`pip install git+https://github.com/ranaroussi/yfinance.git@BRANCH`

E.g.:

`pip install git+https://github.com/ranaroussi/yfinance.git@feature/something`

**Reply by kaustabhray** [reply]
Hi,
I tried the pip route. It broke the basic code for getting historical data.
Also other places....
I used the following command.
`pip install git+https://github.com/ranaroussi/yfinance.git@feature/cookie-and-crumb`

I use conda env.

Screenshot of getting historical data (error)
![image](https://github.com/ranaroussi/yfinance/assets/140444615/ec06fedd-9ea7-4680-b30c-4d405c270eca)

Another screenshot
![image](https://github.com/ranaroussi/yfinance/assets/140444615/1268d985-f002-4bf9-9005-838d3180bb2a)

reverting back to the stable yfinance.

**Reply by kaustabhray** [reply]
I installed yfinance (stable version) and everything works OK now, except the Ticker.info 
 
The screenshot of pip install output.
![image](https://github.com/ranaroussi/yfinance/assets/140444615/07563bbc-12b1-4286-8483-fe9886333f43)

**Comment by ValueRaider** [comment]
# Without PIP

### 1: Download from GitHub:

`git clone https://github.com/ranaroussi/yfinance.git`

Or if a specific branch:

`git clone -b <BRANCH NAME> https://github.com/ranaroussi/yfinance.git`

### 2: Add download location to Python search path

Two different ways, choose one:
- add path to [`PYTHONPATH` environment variable](https://bic-berkeley.github.io/psych-214-fall-2016/using_pythonpath.html)
- add to top of Python file: `import sys ; sys.path.insert(0, "path/to/downloaded/yfinance")`

### 3: Verify
```
import yfinance
print(yfinance)
```
Output should be:

> <module 'yfinance' from '**path/to/downloaded**/yfinance/yfinance/\_\_init\_\_.py'>

If output looks like this then you did step 2 wrong
> <module 'yfinance' from '.../**lib/python3.10/site-packages**/yfinance/\_\_init\_\_.py'>

**Comment by idan6767** [comment]
Hey again,
i tired fixing yfinance "Exception: Yahoo has again changed data format, yfinance now unsure which key(s) is for decryption"
running this line in the termial:   pip install git+https://github.com/ranaroussi/yfinance.git@hotfix

i get this error:

Collecting git+https://github.com/ranaroussi/yfinance.git@hotfix
  Cloning https://github.com/ranaroussi/yfinance.git (to revision hotfix) to c:\users\idan6\appdata\local\temp\pip-req-build-j1mr1u1s
  ERROR: Error [WinError 2] The system cannot find the file specified while executing command git version
ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?

**Reply by paulmcq** [reply]
Just what it says: do you have 'git' installed and in your PATH?
If git is not installed, then install it. (Google 'install git on Windows'.)
If git is not on your PATH, edit your PATH. (Google is your friend.)
Maybe someone who knows Windows can give you better details...

**Comment by ValueRaider** [comment]
superseded by https://github.com/ranaroussi/yfinance/blob/main/CONTRIBUTING.md
